src/v/cluster_link/link.cc

```diff
@@ -89,7 +97,16 @@ ss::future<> link::stop() {
           "Stopping task {} for cluster link {}",
           t->name(),
           _config.name);
-        auto res = co_await stop_task(t.get());
+        auto res_f = co_await ss::coroutine::as_future(stop_task(t.get()));
+        if (res_f.failed()) {
+            vlog(
+              cllog.warn,
+              "Failed to stop task {}: {}",
+              t->name(),
+              res_f.get_exception());
+            continue;
+        }
+        auto res = res_f.get();
```

Comment:
I am not sure if this is the right approach. What happens if one of th task fails to stop and we continue to shutdown the link, the task can access a nullptr